vm/src/format.rs

```diff
@@ -428,6 +428,10 @@ impl FormatSpec {
     }
 
     pub(crate) fn format_int(&self, num: &BigInt) -> Result<String, &'static str> {
+        if self.precision.is_some() {
+            return Err("Precision not allowed in integer format specifier");
+        }
+
```

Comment:
This check should be conducted only if the format type is a kind of integer. That is, the checks should not be executed when the format type is not a kind of integer.